src/pages/home/ReportScreen.js

```diff
@@ -192,18 +191,8 @@ class ReportScreen extends React.Component {
             return null;
         }
 
-        // We create policy rooms for all policies, however we don't show them unless
-        // - It's a free plan workspace
-        // - The report includes guides participants (@team.expensify.com) for 1:1 Assigned
-        if (!Permissions.canUseDefaultRooms(this.props.betas)
-            && ReportUtils.isDefaultRoom(this.props.report)
-            && ReportUtils.getPolicyType(this.props.report, this.props.policies) !== CONST.POLICY.TYPE.FREE
-            && !ReportUtils.hasExpensifyGuidesEmails(lodashGet(this.props.report, ['participants'], []))
-        ) {
-            return null;
-        }
-
-        if (!Permissions.canUsePolicyRooms(this.props.betas) && ReportUtils.isUserCreatedPolicyRoom(this.props.report)) {
+        // Filter out the report that shouldn't be displayed, don't care same reportIDFromRoute, and use default priorityMode
+        if (!ReportUtils.shouldReportBeInOptionList(this.props.report, null, false, this.props.iouReports, this.props.betas, this.props.policies)) {
```

Comment:
Don't we need to pass the real value of `isInGSDMode`? It looks like the logic inside `shouldReportBeInOptionList` changes based on that

```js
    const isInDefaultMode = !isInGSDMode;
    ...
    // All unread chats (even archived ones) in GSD mode will be shown. This is because GSD mode is specifically for focusing the user on the most relevant chats, primarily, the unread ones
    if (isInGSDMode) {
        return isUnread(report);
    }

    // Archived reports should always be shown when in default (most recent) mode. This is because you should still be able to access and search for the chats to find them.
    if (isInDefaultMode && isArchivedRoom(report)) {
        return true;
    }
```